### ISSUE TITLE
Text editor support for browsers without 'beforeinput'

### ISSUE THREAD

**Issue by askdfjlas** [askdfjlas/askdfjlas.github.io#6]
Most notably Firefox, a possible solution is using the 'input' event instead and relying on the DOM sanitization to remove extra characters.

**Comment by askdfjlas** [comment]
The solution I used was as follows:
Keep the original 'beforeinput' handler, it does nothing for browsers which don't support it.
Add an 'input' handler which is identical, except it simulates cancelling the event by immediately sanitizing the text area. The problem with this is that it messes up the caret placement (the text editor is effectively unfocused). Usually, the caret info is always updated prior to insertion, but it's after this sanitization process. So it would result in some nonsense caret info. The solution to this was to conditionally update the caret info, and now for the 'input' event, it would only update if the inputType is 'insertReplacementText'. This was actually the original reason why the caret info was updated during insertion. But I'm assuming that this always places the caret back in the correct position; I'm not sure if it's right or not.
